File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ImagePickerFileProvider.java
package com.esafirm.imagepicker.helper;

import androidx.core.content.FileProvider;

import com.esafirm.imagepicker.features.ImagePickerComponentHolder;

public class ImagePickerFileProvider extends FileProvider {
    @Override
    public boolean onCreate() {
        ImagePickerComponentHolder.getInstance().init(getContext());
        return super.onCreate();
    }
}

File: imagepicker/src/main/java/com/esafirm/imagepicker/helper/ImagePickerFileProvider.kt
package com.esafirm.imagepicker.helper

import androidx.core.content.FileProvider
import com.esafirm.imagepicker.features.DefaultImagePickerComponents
import com.esafirm.imagepicker.features.ImagePickerComponentsHolder

class ImagePickerFileProvider : FileProvider() {
    override fun onCreate(): Boolean {
        ImagePickerComponentsHolder.setInternalComponent(DefaultImagePickerComponents(context!!))
        return super.onCreate()
    }
}